forms-flow-api/src/formsflow_api/services/authorization.py

```diff
@@ -3,19 +3,203 @@
 import datetime
 from typing import Dict, List
 
+from flask import current_app
 from formsflow_api_utils.exceptions import BusinessException
 from formsflow_api_utils.utils.user_context import UserContext, user_context
+from sqlalchemy.orm.attributes import flag_modified
 
 from formsflow_api.constants import BusinessErrorCode
-from formsflow_api.models import Application, Authorization, AuthType
+from formsflow_api.models import Application, Authorization, AuthType, db
 from formsflow_api.schemas import ApplicationSchema
+from formsflow_api.services.external.analytics_api import RedashAPIService
 
 application_schema = ApplicationSchema()
 
 
 class AuthorizationService:
     """This class manages authorization service."""
 
+    def check_and_update_name_mismatch(  # pylint: disable=too-many-arguments, too-many-positional-arguments
+        self,
+        auth_name: str,
+        dashboard_id: str,
+        dashboard: Dict[str, str],
+        auth_record: Authorization,
+        updates_needed: List[Dict],
+        result: List[Dict],
+    ):
+        """Check if the name in auth record matches the dashboard name and update if needed."""
+        if auth_name != dashboard["name"]:
```

Comment:
How are we handling status changes like `enabled / disabled / archived` etc from analytics results?